### PR TITLE
Don't ignore the fee when it's explicitly specified.

### PR DESCRIPTION
**Pull request by dskloet**
Make sure a specified transaction fee and outputs add up to the sum of the inputs.

### CODE REVIEW

**Comment by dskloet** [comment]
I'm looking at how the coverage could have decreased.

**Comment by maraoz** [comment]
LGTM

**Comment by dskloet** [comment]
I found a problem when investigating the coverage. I'll write a test and fix it. Is there a way to amend the pull request or do I have to make a new pull request?

**Comment by martindale** [comment]
@dskloet just push new commits to the same branch and it'll automatically associate them with this pull request!  :)

**Comment by dskloet** [comment]
Thanks!
The problem wasn't caught by the tests because a test that expected a FeeError passed because it got a FeeError for a different reason than intended. I'm fixing it by introducing some different kinds of fee errors to distinguish the different cases.

I've pushed the fix.

**Comment by dskloet** [comment]
FYI: This pull requests addresses part of issue #1174.

File: lib/errors/spec.js [hunk 1]
```diff
@@ -85,8 +85,14 @@ module.exports = [{
...
-    message: 'Fees are not correctly set {0}',
+    name: 'SmallFeeError',
+    message: 'Fee is too small: {0}',
+  }, {
+    name: 'LargeFeeError',
+    message: 'Fee is too large: {0}',
+  }, {
+    name: 'DifferentFeeError',
+    message: 'Unspent value is different from specified fee: {0}',
   }, {
```
Comment (braydonf) [inline lib/errors/spec.js]: Perhaps we could make these additional errors part of a general `FeeError`:

```js
if (err instanceof errors.FeeError) {
  //...
}
```
And if interested in a more specific exception:
```js
if (err instanceof errors.FeeError.TooSmall) {
  //...
}
```

**Comment by braydonf** [comment]
Is there a use case for disabling the check if fees are different? 

If so, the related [Transaction serialization docs](https://github.com/bitpay/bitcore/blob/master/docs/transaction.md#serialization-checks) need to be updated with the additional serialization option.

**Comment by dskloet** [comment]
The check wasn't there before so I'm mostly just being defensive. Would you prefer I remove the ability to turn it off or add it to the docs?

**Comment by dskloet** [comment]
I realized that if it can be disabled, I also need to add a test that it can be disabled.

File: lib/errors/spec.js [hunk 2]
```diff
@@ -85,8 +85,18 @@ module.exports = [{
     name: 'InvalidSatoshis',
     message: 'Output satoshis are invalid',
   }, {
-    name: 'FeeError',
-    message: 'Fees are not correctly set {0}',
+    name: 'Fee',
```
Comment (braydonf) [inline lib/errors/spec.js]: For backwards compatibility, we can keep this as `FeeError`

**Comment by braydonf** [comment]
If the fee is specified and the actual fee would differ, I can't think of a use case to disable the check, as by not specifying an exact fee would be the means to do such. (I'll verify though).

**Comment by dskloet** [comment]
Makes sense. Let me know when you've verified it and I'll remove it.

**Comment by braydonf** [comment]
Yep, works as expected.

**Pull request merged**